services/api-server/src/simcore_service_api_server/services/log_streaming.py

```diff
@@ -100,6 +98,11 @@ async def deregister(self, job_id: JobID):
         )
         del self._log_streamers[job_id]
 
+    def get_log_queue_sizes(self) -> dict[JobID, int]:
+        keys: list[JobID] = list(self._log_streamers.keys())
+        values: list[int] = list(map(lambda q: q.qsize(), self._log_streamers.values()))
+        return dict(zip(keys, values))
```

Comment:
I did it like this to avoid python for loops for performance. That could be a misjudgement though. Point is that this is done in a blocking manner, so performance is somewhat critical here.

Reply:
whaaaaaat? :-D 
Can you demonstrate me that your approach is more performant than mine. Bets accepted! :-D

```python
import timeit
import random
from typing import NamedTuple, Dict

# Mock classes to simulate the environment
class JobID(NamedTuple):
    id: int

class QueueMock:
    def __init__(self):
        self.size = random.randint(1, 100)
    
    def qsize(self):
        return self.size

class PerformanceTest:
    def __init__(self, num_elements: int):
        self._log_streamers: Dict[JobID, QueueMock] = {
            JobID(id=i): QueueMock() for i in range(num_elements)
        }
    
    def solution1(self):
        return {k: v.qsize() for k, v in self._log_streamers.items()}
    
    def solution2(self):
        keys = list(self._log_streamers.keys())
        values = list(map(lambda q: q.qsize(), self._log_streamers.values()))
        return dict(zip(keys, values))

# Number of elements in the mock dictionary
num_elements = 1000
test_obj = PerformanceTest(num_elements)

# Time both solutions
time_solution1 = timeit.timeit(test_obj.solution1, number=10000)
time_solution2 = timeit.timeit(test_obj.solution2, number=10000)

print(time_solution1, time_solution2)
```

produced
```cmd
(1.8377870200000004, 2.2238771529999966)
```

Reply:
Corollary 2:
- should wait for the bets before providing the result

Reply:
NOTE1: Introspections are special loops in python. Do not underestimate the
NOTE2: py3.X aim to improve more and more those bottlenecks. Do not pre-optimize since typically you can get it resolved with an update "for free"